test_runner/src/main/kotlin/ftl/reports/MatrixResultsReport.kt

```diff
@@ -48,14 +49,23 @@ object MatrixResultsReport : IReport {
             }
 
             if (matrices.map.isNotEmpty()) {
-                writer.println("More details are available at [${matrices.map.values.first().webLinkWithoutExecutionDetails}]")
-                writer.println(matrices.map.values.toList().asPrintableTable())
+                val savedMatrices = matrices.map.values
+                writer.println(savedMatrices.toList().asPrintableTable())
+                savedMatrices.printMatricesLinks(writer)
             }
 
             return writer.toString()
         }
     }
 
+    private fun Collection<SavedMatrix>.printMatricesLinks(writer: StringWriter) {
```

Comment:
Here is my proposal for formatting. I am just not a big fan of multiple `?.` chain calls. Of course, it's not change request. Just let me know what do you think
```suggestion
    private fun Collection<SavedMatrix>.printMatricesLinks(writer: StringWriter) = this
        .filter { it.failed() }
        .takeIf { it.isNotEmpty() }
        ?.run { this
            .also { writer.println("More details are available at:") }
            .map { it.webLinkWithoutExecutionDetails }
            .forEach { writer.println(it) }
        }
```

Reply:
I personally preferer to avoid nested operations in block, that's why I changed it to `?.` chain (previous version looks like yours :) )

Reply:
Got your point. We can still make it simpler skip one `map` operation and just invoke `forEach { writer.println(it.webLinkWithoutExecutionDetails) }`

Reply:
applied your suggestion, because of new line comment :)